File: library/src/main/java/com/davemorrissey/labs/subscaleview/provider/UriInputProvider.java
package com.davemorrissey.labs.subscaleview.provider;

import android.content.ContentResolver;
import android.content.Context;
import android.net.Uri;

import java.io.IOException;
import java.io.InputStream;

public class UriInputProvider implements InputProvider {

    public final Uri uri;
    private final ContentResolver resolver;

    public UriInputProvider(Context context, Uri uri) {
        this.resolver = context.getContentResolver();
        this.uri = uri;
    }

    @Override
    public InputStream openStream() throws IOException {
        return resolver.openInputStream(uri);
    }
}

File: library/src/main/java/com/davemorrissey/labs/subscaleview/provider/UriInputProvider.kt
package com.davemorrissey.labs.subscaleview.provider

import android.content.ContentResolver
import android.content.Context
import android.net.Uri
import java.io.IOException
import java.io.InputStream

class UriInputProvider(
    context: Context,
    private val uri: Uri,
) : InputProvider {

    private val resolver: ContentResolver = context.contentResolver

    @Throws(IOException::class)
    override fun openStream(): InputStream? {
        return resolver.openInputStream(uri)
    }
}